apps/authorization/tests/test_data_access_grant.py

```diff
@@ -249,3 +255,116 @@ def test_permission_deny_on_app_or_org_disabled(self):
+                                     user_hicn_hash='255e178537ed3bc486e6a7195a47a82a2cd6f46e911660fe9775f6e00000000' + str(cnt),
+                                     user_mbi_hash='987654321aaaa11111aaaa195a47a82a2cd6f46e911660fe9775f6e00000000' + str(cnt))
+            # create AC
+            AccessToken.objects.create(
+                token="existingtoken1Synth" + str(cnt),
+                user=user,
+                application=application1,
+                expires=timezone.now() + timedelta(seconds=10),
+            )
+
+        # Create 3x Real (positive FHIR_ID) users and access tokens for applicaiton2
+        for cnt in range(3):
+            user = self._create_user('joesmith' + str(cnt), 'password',
+                                     first_name='Joe' + str(cnt),
+                                     last_name='Smith',
+                                     email='joe' + str(cnt) + '@smith.net',
+                                     fhir_id='1000000000000' + str(cnt),
+                                     user_hicn_hash='509e178537ed3bc486e6a7195a47a82a2cd6f46e911660fe9775f6e00000000' + str(cnt),
+                                     user_mbi_hash='9076543217ed3bc486e6a7195a47a82a2cd6f46e911660fe9775f6e00000000' + str(cnt))
+            # create AC
+            AccessToken.objects.create(
+                token="existingtoken2Real" + str(cnt),
+                user=user,
+                application=application2,
+                expires=timezone.now() + timedelta(seconds=10),
+            )
+
+        # Create 9x Synthetic (negative FHIR_ID) users and access tokens for application2
+        for cnt in range(9):
+            user = self._create_user('joedoe' + str(cnt), 'password',
+                                     first_name='Joe' + str(cnt),
+                                     last_name='Doe',
+                                     email='joe' + str(cnt) + '@doe.net',
+                                     fhir_id='-1000000000000' + str(cnt),
+                                     user_hicn_hash='505e178537ed3bc486e6a7195a47a82a2cd6f46e911660fe9775f6e00000000' + str(cnt),
+                                     user_mbi_hash='907654321aaaa11111aaaa195a47a82a2cd6f46e911660fe9775f6e00000000' + str(cnt))
+            # create AC
+            AccessToken.objects.create(
+                token="existingtoken2Synth" + str(cnt),
+                user=user,
+                application=application2,
+                expires=timezone.now() + timedelta(seconds=10),
+            )
+
+        # Assert crosswalks counts
+        self.assertEqual("{'synthetic': 16, 'real': 8}", str(check_crosswalks()))
+
+        # This creates grants from access tokens. Does not test creation on approval (this is tested elsewhere).
+        with transaction.atomic():
+            update_grants()
+
+        # Assert check_grants (all grants)
+        self.assertEqual("{'unique_tokens': 24, 'grants': 24}", str(check_grants()))
```

Comment:
Ran in to a case in Jenkins CBC where the assert failed with a count of 21 instead of 24.  Running it again without any code changes got a successful count of 24. 

I added the `transaction.atomic()` context manager around update_grants() that I believe will prevent that issue.